cou/apps/auxiliary.py

```diff
@@ -53,31 +53,29 @@ def is_valid_track(self, charm_channel: str) -> bool:
             logger.debug("'%s' has been installed from the charm store", self.name)
             return True
 
-        track = self._get_track_from_channel(charm_channel)
-        return (self.charm, self.series, track) in TRACK_TO_OPENSTACK_MAPPING
+        current_track = self._get_track_from_channel(charm_channel)
+        possible_tracks = OPENSTACK_TO_TRACK_MAPPING.get(
+            (self.charm, self.series, self.current_os_release.codename), []
+        )
+        return (
+            self.charm,
+            self.series,
+            current_track,
+        ) in TRACK_TO_OPENSTACK_MAPPING and len(possible_tracks) > 0
 
     @property
-    def possible_current_channels(self) -> list[str]:
-        """Return the possible current channels based on the series and current OpenStack release.
+    def possible_current_channel(self) -> str:
+        """Return the possible current channel.
 
-        :return: The possible current channels for the application.
-        :rtype: list[str]
-        :raises ApplicationError: When cannot find tracks.
+        Possible current channel is based in the series and in the current OpenStack release of
+        the application.
+        :return: The possible current channel of the application. E.g: "3.9/stable"
+        :rtype: str
         """
-        tracks = OPENSTACK_TO_TRACK_MAPPING.get(
-            (self.charm, self.series, self.current_os_release.codename)
-        )
-        if tracks:
-            return [f"{track}/stable" for track in tracks]
-
-        raise ApplicationError(
-            (
-                f"Cannot find a suitable '{self.charm}' charm channel for "
-                f"{self.current_os_release.codename} on series '{self.series}'. "
-                "Please take a look at the documentation: "
-                "https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html"
-            )
+        *_, track = OPENSTACK_TO_TRACK_MAPPING.get(
+            (self.charm, self.series, self.current_os_release.codename), []
```

Comment:
Thanks for pointing this. Because `is_valid_track`already checks that exists some track, we can access here directly without the need of using `.get` or worry with keyerror

Reply:
In that case let's remove the fallback or even `.get` to avoid confusion.